flags.go

```diff
@@ -22,6 +22,19 @@ var (
 		Usage:    "AVS Operator state retriever address",
 		EnvVar:   envVarPrefix + "OPERATOR_STATE_RETRIEVER_ADDR",
 	}
+	ServiceManagerAddrFlag = cli.StringFlag{
+		Name:     "service-manager-addr",
+		Required: true,
+		Usage:    "AVS Service Manager address",
+		EnvVar:   envVarPrefix + "SERVICE_MANAGER_ADDR",
+	}
+	DontUseAllocationManagerFlag = cli.BoolFlag{
+		Name: "dont-use-allocation-manager",
+		Usage: "If set to true, does not use the AllocationManager core contract to handle operator registration/deregistration and stake allocation/deallocation." +
+			"Only set this to true if you are using the legacy AVSDirectory core contract system. For new deployments not utilizing the M2 flows and AVSs implementing slashing, " +
+			"this should remain false to enable proper functionality.",
```

Comment:
On the SDK side, this flag only disables a call to `DelegationManager.allocationManager` in the `chainio` constructors, which failed because the `AllocationManager` wasn't deployed yet on mainnet. It should be used when the EigenLayer deployment is pre-slashing (i.e. mainnet right now), and doesn't depend on the registration flow the AVS uses.